File: client-side/src/app/shared/models/blog.ts
import { Doctor } from './doctor';

export interface Blog {
  id: number;
  title: string;
  slug: string;
  description: string;
  photo: string;
  doctor?: Doctor;
  video?: any;
  commentCount: number;
  addedDate: Date;
}

export class BlogB implements Blog {
  id: number;
  title: string;
  slug: string;
  description: string;
  photo: string;
  doctor?: Doctor;
  video?: any;
  commentCount: number;
  addedDate: Date;

  constructor(
    id: number,
    title: string,
    slug: string,
    description: string,
    photo: string,
    doctor?: Doctor,
    video?: any,
  ) {
    this.id = id;
    this.title = title;
    this.slug = slug;
    this.description = description;
    this.photo = photo;
    this.doctor = doctor;
    this.video = video;
  }
}

File: server-side/Core/Models/Blog.cs
namespace Core.Models
{
    public class Blog : BaseEntity
    {
        public string Title { get; set; }
        public string Slug { get; set; }
        public string Video { get; set; }
        public string Description { get; set; }
        public string Photo { get; set; }

        public int? DoctorId { get; set; }
        public Doctor Doctor { get; set; }
    }
}